src/app/asset-page/asset-page.component.ts

```diff
@@ -453,6 +455,27 @@ export class AssetPage implements OnInit, OnDestroy {
     this._group.hasPrivateGroups()
   } // OnInit
 
+  get showDownloadDropdown() {
+    const isLoggedIn = this.user && this.user.isLoggedIn;
+    const publicDownload = this.assets && this.assets[0] && this.assets[0].publicDownload;
+    const downloadLink = this.assets && this.assets[0] && this.assets[0].downloadLink;
+    const userStatus = this.user && this.user.status;
+
+    if (!downloadLink) {
+      return false;
+    }
+
+    if (!this.requireLoginForAdlDownload) {
+      if (!publicDownload) {
+        return userStatus || isLoggedIn;
+      }
+
+      return true;
+    }
+
+    return !!(isLoggedIn || publicDownload);
+  }
+
```

Comment:
This already reads a lot better than when it lives in the template! Wondering if we can reduce it even further. If I am following things correctly I think that `publicDownload` always seems to resolve be `true` (assuming we actually have the `downloadLink`. Wondering if we can move that upwards to simplify things.

Does this seem equivalent?
```if(!downloadLink) {
	return false;
}

if(publicDownload) {
	return true;
}

if(!this.requireLoginForAdlDownload) {
	return userStatus;
}

return isLoggedIn;```

Reply:
Refactored in f30adbf